### PR TITLE
[incubator-kie-issues#1473] Add withTransactional method to  DependencyInjectionAnnotator

### LINKED ISSUE #1473: Add transaction annotations to processes endpoints
Add the `transaction` annotation in endpoints in here
https://github.com/apache/incubator-kie-kogito-runtimes/tree/main/kogito-codegen-modules/kogito-codegen-processes/src/main/resources/class-templates
and verify that the data sources are working as expected with the data source pgsql.

### PR DESCRIPTION
**Pull request by gitgabrio**
Fixes https://github.com/apache/incubator-kie-issues/issues/1473

Needed by https://github.com/apache/incubator-kie-kogito-runtimes/pull/3671

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request and downstream checks</b>  
  - Push a new commit to the PR. An empty commit would be enough.

- for a <b>full downstream build</b>
  - for <b>github actions</b> job: add the label `run_fdb`

- for <b>Jenkins PR check only</b>
  - If you are an ASF committer for KIE podling, login to Jenkins (https://ci-builds.apache.org/job/KIE/job/drools/), go to the specific PR job, and click on `Build Now` button.
</details>

### CODE REVIEW

File: drools-model/drools-codegen-common/src/main/java/org/drools/codegen/common/di/impl/SpringDependencyInjectionAnnotator.java [hunk 1]
```diff
@@ -154,19 +161,22 @@ public String emitterType(String dataType) {
...
+                                                          new StringLiteralExpr("${" + configKey + ":#{null}}")));
         return node;
     }
 
     @Override
     public <T extends NodeWithAnnotations<?>> T withConfigInjection(T node, String configKey, String defaultValue) {
-        node.addAnnotation(new SingleMemberAnnotationExpr(new Name("org.springframework.beans.factory.annotation.Value"), new StringLiteralExpr("${" + configKey + ":" + defaultValue + "}")));
+        node.addAnnotation(new SingleMemberAnnotationExpr(new Name("org.springframework.beans.factory.annotation" +
+                                                                           ".Value"),
+                                                          new StringLiteralExpr("${" + configKey + ":" + defaultValue + "}")));
```
Comment (mariofusco) [inline drools-model/drools-codegen-common/src/main/java/org/drools/codegen/common/di/impl/SpringDependencyInjectionAnnotator.java]: Same as above.

File: drools-model/drools-codegen-common/src/main/java/org/drools/codegen/common/di/impl/SpringDependencyInjectionAnnotator.java [hunk 2]
```diff
@@ -154,19 +161,22 @@ public String emitterType(String dataType) {
 
     @Override
     public <T extends NodeWithAnnotations<?>> T withConfigInjection(T node, String configKey) {
-        node.addAnnotation(new SingleMemberAnnotationExpr(new Name("org.springframework.beans.factory.annotation.Value"), new StringLiteralExpr("${" + configKey + ":#{null}}")));
+        node.addAnnotation(new SingleMemberAnnotationExpr(new Name("org.springframework.beans.factory.annotation" +
+                                                                           ".Value"),
+                                                          new StringLiteralExpr("${" + configKey + ":#{null}}")));
```
Comment (mariofusco) [inline drools-model/drools-codegen-common/src/main/java/org/drools/codegen/common/di/impl/SpringDependencyInjectionAnnotator.java]: Very minor thing, but I'm not understanding the reason of this change.

**Review by mariofusco — APPROVED** [review]
That's ok for me, I just noticed a few weird rewriting, very likely made by the formatter, that's however is introducing some unnecessary String concatenations, can we get rid of them?

**Review by pefernan — APPROVED** [review]

**Review by elguardian — APPROVED** [review]
LGTM

**Review by yesamer — APPROVED** [review]

**Pull request merged**